src/server/pfs/server/api_server.go

```diff
@@ -384,97 +387,87 @@ type modifyFileSource interface {
 	Recv() (*pfs.ModifyFileRequest, error)
 }
 
-func (a *apiServer) modifyFile(ctx context.Context, uw *fileset.UnorderedWriter, server modifyFileSource, req *pfs.ModifyFileRequest) (int64, error) {
-	var bytesRead int64
-	for {
-		// TODO Validation.
-		if req != nil && req.Modification != nil {
-			switch mod := req.Modification.(type) {
-			case *pfs.ModifyFileRequest_PutFile:
-				var err error
-				var n int64
-				switch mod.PutFile.Source.(type) {
-				case *pfs.PutFile_RawFileSource:
-					n, err = putFileRaw(uw, server, mod.PutFile)
-				case *pfs.PutFile_TarFileSource:
-					n, err = putFileTar(uw, server, mod.PutFile)
-				case *pfs.PutFile_UrlFileSource:
-					n, err = putFileURL(ctx, uw, mod.PutFile)
-				}
-				if err != nil {
-					return bytesRead, err
-				}
-				bytesRead += n
-			case *pfs.ModifyFileRequest_DeleteFile:
-				if err := deleteFile(uw, mod.DeleteFile); err != nil {
-					return bytesRead, err
-				}
-			case *pfs.ModifyFileRequest_CopyFile:
-				cf := mod.CopyFile
-				if err := a.driver.copyFile(ctx, uw, cf.Dst, cf.Src, cf.Append, cf.Tag); err != nil {
-					return bytesRead, err
-				}
-			}
+// modifyFile reads from a modifyFileSource until io.EOF and writes changes to an UnorderedWriter.
+// SetCommit messages will result in an error.
+func (a *apiServer) modifyFile(ctx context.Context, uw *fileset.UnorderedWriter, server modifyFileSource) (int64, error) {
```

Comment:
We should rewrite this to just apply each message as they come in to the buffering layer. We can get rid of the majority of this logic if we do that.

Reply:
You want to force the client to provide the path and tag for each message?

Reply:
Yeah, that is what I thought we agreed on? We make each message an independent operation so we don't need to manage streams on the client or server side.

Reply:
I guess we can always add the other behavior later.